### PR TITLE
[CPM-166] Trigger toast on Save Hotkey

### PR DESCRIPTION
**Pull request by FocalChord**
## Changes

- Ctrl-S or Command-S will trigger a toast saying you don't need to save :)
- Adding autosave

## Checklist

- [x] Have tests been written?
- [x] Is the UI consistent?

## Requests
- [x] Can somebody check the styling/CSS?

Addresses [CPM-166]

[CPM-166]: https://momentum-agile.atlassian.net/browse/CPM-166

### CODE REVIEW

**Comment by daltonlim** [comment]
@pull-dog up

**Review by daltonlim — APPROVED** [review]
LGTM 🥇

**Review by Richardng12 — APPROVED** [review]
lgtm!!

File: client/src/pages/Courses/useCourses.js [hunk 1]
```diff
@@ -70,7 +70,9 @@ const useCourses = () => {
         return CoursePlannerClient.getSpecificCourse(subject, courseNumber);
     };
 
-    useEffect(() => fetchAllCourses(), []);
+    useEffect(() => {
```
Comment (vee-christella) [inline client/src/pages/Courses/useCourses.js:73]: Is there a reason for this change?

File: client/src/pages/Plan/Header.js [hunk 2]
```diff
@@ -31,8 +43,21 @@ const Header = ({ name, programme, planName, setPlanName }) => {
...
+                <Flex align="center" justify="center" direction={"column"}>
+                    <InlineEdit
+                        title="Plan name"
+                        value={editPlanName}
+                        onSubmit={() => setPlanName(editPlanName)}
+                        onChange={(e) => setEditPlanName(e)}
+                    />
+                    <Text>
+                        {" "}
+                        Last saved{" "}
```
Comment (vee-christella) [inline client/src/pages/Plan/Header.js:55]: Love this! ❤️

File: client/src/pages/Plan/Plan.js [hunk 3]
```diff
@@ -41,18 +38,31 @@ const Plan = () => {
...
+        lastSaveDate,
     } = usePlan();
+
+    const { name, numYears, startYear, courseAllocations } = plan;
+
     const [searchTerm, setSearchTerm] = useState("");
     const [note, setNote] = useState("");
+    const toast = useToast();
+
+    useHotkeys("cmd+s, ctrl+s", (e) => {
```
Comment (vee-christella) [inline client/src/pages/Plan/Plan.js:56]: Thanks for accounting for both cmd and ctrl 😄

**Review by vee-christella — APPROVED** [review]
Other than changes I've requested via messenger chat, LGTM

File: client/src/pages/Courses/useCourses.js (hunk 1 above)
Comment (daltonlim) [reply client/src/pages/Courses/useCourses.js:73]: breaks premaster if u dont have it lmao

**Pull request merged**